### PR TITLE
fix(integ-runner): catch snapshot errors, treat `--from-file` as command-line

### PR DESCRIPTION
**Pull request by rix0rrr**
Snapshot errors
---------------

The constructor of `IntegSnapshotRunner` calls `loadManifest()`, which
on my computer happened to fail and stopped the entire test suite
because this error happened outside the `try/catch` block. Same for
the integ runner itself.

Move it inside the try/catch block, needed a bit of refactoring to
make sure we could still get at the test name.

`--from-file`
-------------

Instead of having `--from-file` require a JSON file with its own
structure, interpret it as a text file which gets treated exactly
the same as the `[TEST [..]]` arguments on the command line.

This still allows for the `--exclude` behavior by setting that flag
on the command-line.

Also be very liberal on the pattern (file name or test name or display
name) that we accept, encoded in the `IntegTest.matches()` class.

Refactoring
-----------

Moved the logic around determining test names and directories into a
class (`IntegTest`) which is a convenience class on top of a static data
record (`IntegTestInfo`). The split between data and logic is so that we
can pass the data to worker threads where we can hydrate the helper
class on top again. I tried to be consistent: in memory, all the fields are 
with respect to `process.cwd()`, so valid file paths in the current
process. Only when they are passed to the CLI wrapper are the paths
made relative to the CLI wrapper directory.

Print snapshot validations that are running for a long time (1 minute).
This helps diagnose what is stuck, if anything is. On my machine, it
was tests using Docker because there was some issue with it, and this
lost me a day. Also change the test reporting formatting slightly.

--------------

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/integ-runner/lib/runner/runner-base.ts [hunk 1]
```diff
@@ -157,8 +197,8 @@ export abstract class IntegRunner {
         ...options.env,
       },
     });
-    this.cdkOutDir = options.integOutDir ?? `${CDK_OUTDIR_PREFIX}.${testName}`;
-    this.cdkApp = `node ${parsed.base}`;
+    this.cdkOutDir = options.integOutDir ?? `${CDK_OUTDIR_PREFIX}.${this.testName}`;
```
Comment (corymhall) [inline packages/@aws-cdk/integ-runner/lib/runner/runner-base.ts:200]: `this.testName` needs to be the `baseTestName` from `testNameFromFile`

Comment (corymhall) [reply packages/@aws-cdk/integ-runner/lib/runner/runner-base.ts:200]: And actually can you just add `baseTestName` to the `TestNameParts`. Maybe call it `stableTestName`? The `testName` (really the `testDisplayName`) is something that changes depending on where you run the test from, but we also need a stable test name that _doesn't_ change to use [here](https://github.com/aws/aws-cdk/blob/777953106ac550b058fdaa3ccde25b62be07defa/packages/%40aws-cdk/integ-runner/lib/runner/integ-test-suite.ts#L159)

Comment (rix0rrr) [reply packages/@aws-cdk/integ-runner/lib/runner/runner-base.ts:200]: Yes I'm running into some trouble with the other change I'm trying to make here as well. They seem to come together.

**Review by corymhall — APPROVED** [review]
Looks great! Love these improvements ❤️

**Pull request merged**